scripts/run-e2e.ts

```diff
@@ -274,13 +274,18 @@ if (frameworkArgs.length > 0) {
 const perform = () => {
   const limit = pLimit(1);
   const narrowedConfigs = Object.values(e2eConfigs);
-  const [a, b] = [+process.env.CIRCLE_NODE_INDEX || 0, +process.env.CIRCLE_NODE_TOTAL || 1];
-  const step = Math.ceil(narrowedConfigs.length / b);
-  const offset = step * a;
-
-  const list = narrowedConfigs.slice().splice(offset, step);
-
-  logger.info(`📑 Assigning jobs ${list.map((c) => c.name).join(', ')} to node ${a} (on ${b})`);
+  const nodeIndex = +process.env.CIRCLE_NODE_INDEX || 0;
+  const numberOfNodes = +process.env.CIRCLE_NODE_TOTAL || 1;
+
+  const list = narrowedConfigs.filter((_, index) => {
```

Comment:
I wanted to do that for a while now ^^
This logic is shared in other runners, chromatic and examples if I'm right

Reply:
https://github.com/storybookjs/storybook/pull/10989